backend/typescript/services/implementations/camperService.ts

```diff
@@ -1264,118 +1265,83 @@ class CamperService implements ICamperService {
-            ],
-            latePickup: [
-              "Thu Sep 07 2023 21:30:00 GMT+0000 (Coordinated Universal Time)",
-            ],
-            registrationDate:
-              "Wed Jan 04 2023 07:17:18 GMT+0000 (Coordinated Universal Time)",
-            hasPaid: true,
-            chargeId:
-              "cs_test_b1ynp34YOUVYdI6Fn6Xq1q8Ka2kCgM6TlVoWy1vPbZpiSfTPY0rrDwAtD9",
-            charges: {
-              camp: 160,
-              earlyDropoff: 19,
-              latePickup: 19,
-            },
-            refundCode:
-              "cs_test_b1ynp34YOUVYdI6Fn6Xq1q8Ka2kCgM6TlVoWy1vPbZpiSfTPY0rrDwAtD8",
-            dates: [
-              "Thu Sep 07 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-              "Sun Sep 03 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-              "Mon Sep 04 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-              "Tue Sep 05 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-            ],
-          },
-          {
-            id: "63b527fed9c42a8fd5ea55e6",
-            campSession: "63b01de0c71ee220ef84daee",
-            earlyDropoff: [
-              "Thu Sep 14 2023 14:30:00 GMT+0000 (Coordinated Universal Time)",
-            ],
-            latePickup: [
-              "Thu Sep 14 2023 21:30:00 GMT+0000 (Coordinated Universal Time)",
-            ],
-            registrationDate:
-              "Wed Jan 09 2023 07:17:18 GMT+0000 (Coordinated Universal Time)",
-            hasPaid: true,
-            chargeId:
-              "cs_test_b1ynp34YOUVYdI6Fn6Xq1q8Ka2kCgM6TlVoWy1vPbZpiSfTPY0rrDwAtD9",
-            charges: {
-              camp: 160,
-              earlyDropoff: 19,
-              latePickup: 19,
-            },
-            refundCode:
-              "cs_test_b1ynp34YOUVYdI6Fn6Xq1q8Ka2kCgM6TlVoWy1vPbZpiSfTPY0rrDwAtD8",
-            dates: [
-              "Thu Sep 14 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-              "Sun Sep 10 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-              "Mon Sep 11 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-              "Tue Sep 12 2023 17:00:00 GMT+0000 (Coordinated Universal Time)",
-            ],
-          },
-        ],
-      },
-    ];
+      // need a camp session to get camp id to get start and end times from camp
+      campSession = await MgCampSession.findById(campers[0].campSession);
+      if (!campSession) {
+        throw new Error(`Camp session with ID ${campers[0].campSession} not found.`);
+      }
+
+      const camp = await MgCamp.findById(campSession.camp);
+      if (!camp) {
+        throw new Error(`Camp with ID ${campSession.camp} not found.`);
+      }
+
+      for (let i = 0; i < campers.length; i++) {
+        camper = campers[i];
+        // eslint-disable-next-line no-await-in-loop
+        campSession = await MgCampSession.findById(campers[i].campSession);
+        key = campers[i].firstName + campers[i].lastName;
```

Comment:
Currently registration does not allow registering two campers with same name in one checkout session. It only registers the first one and ignores any duplicates.

Using camper's firstName + lastName should be unique key since all campers returned [here](https://github.com/uwblueprint/focus-on-nature/pull/276/commits/09c2133149396a81c75260f0041715c810d6243c#diff-da500917cfe8a24b6558fb1d2419cfca87e7716ee3ab04e08f0a20823dd68821R1279) `const campers: Camper[] = await MgCamper.find({ refundCode });` are registered under same checkout session.

Cannot use camper id since they are different for every camper in campers regardless of their firstName and lastName.